addon/PlacesProvider.js

```diff
@@ -749,6 +749,53 @@ Links.prototype = {
   }),
 
   /**
+   * Gets bookmarks, sorted in descending order by edit date, by default limited to 3 items.
+   *
+   * @returns {Promise} Returns a promise with the most recent bookmarks.
+   */
+  getBookmarks: Task.async(function*(options = {limit: 3}) {
+    let {limit} = options;
+
+    const columns = [
+      "bookmarkDateCreated",
+      "bookmarkGuid",
+      "bookmarkId",
+      "bookmarkTitle",
+      "frecency",
+      "guid",
+      "type",
+      "lastModified",
+      "lastVisitDate",
+      "title",
+      "url",
+      "visitCount"
+    ];
+
+    let sqlQuery = `SELECT p.url as url,
+                           p.guid as guid,
+                           p.title as title,
+                           p.frecency as frecency,
+                           p.visit_count as visitCount,
+                           p.last_visit_date / 1000 as lastVisitDate,
+                           b.id as bookmarkId,
+                           "bookmark" as type,
+                           b.guid as bookmarkGuid,
+                           b.title as bookmarkTitle,
+                           b.lastModified / 1000 as lastModified,
+                           b.dateAdded / 1000 as bookmarkDateCreated
+                    FROM moz_bookmarks b, moz_places p
+                    WHERE type = :type
+                    AND b.fk = p.id
+                    AND p.last_visit_date IS NOT NULL
+                    ORDER BY b.lastModified DESC
+                    LIMIT ${limit}`;
+
+    let result = yield this.executePlacesQuery(sqlQuery, {columns, params: {type: Bookmarks.TYPE_BOOKMARK}});
+
+    return result;
```

Comment:
Comparing to some of the other queries, they have this after the query instead of returning the query results directly. Do we not need this?

```
links = yield this._addFavicons(links);
return this._processLinks(links);
```